### PR TITLE
Only lint call expressions and external module references in no-require-imports rule

### LINKED ISSUE #816: no-require-imports false positive
```ts
import TestUtils = React.addons.TestUtils;
```

should not report a failure with the `no-require-imports` rule.

### PR DESCRIPTION
**Pull request by adidahiya**
Fixes #816.

### CODE REVIEW

File: src/rules/noRequireImportsRule.ts [hunk 1]
```diff
@@ -26,27 +26,20 @@ export class Rule extends Lint.Rules.AbstractRule {
...
-
-    private handleDeclaration(decl: ts.VariableDeclaration)  {
-        // make sure the RHS is a call expression.
-        const call = <ts.CallExpression> (decl.initializer);
-        if (call && call.arguments && call.expression) {
-            const callExpressionText = call.expression.getText(this.getSourceFile());
-            if (callExpressionText === "require") {
-                this.addFailure(this.createFailure(decl.getStart(), decl.getWidth(), Rule.FAILURE_STRING));
-            }
+        const {moduleReference} = node;
```
Comment (JKillian) [inline src/rules/noRequireImportsRule.ts]: Do we normally use spaces here? `{ moduleReference }`? Not sure

File: src/rules/noRequireImportsRule.ts [hunk 2]
```diff
@@ -26,27 +26,20 @@ export class Rule extends Lint.Rules.AbstractRule {
 }
 
 class NoRequireImportsWalker extends Lint.RuleWalker {
-    public visitVariableStatement(node: ts.VariableStatement) {
-        const declarations = node.declarationList.declarations;
-        for (let decl of declarations) {
-            this.handleDeclaration(decl);
+    public visitCallExpression(node: ts.CallExpression) {
```
Comment (JKillian) [inline src/rules/noRequireImportsRule.ts]: Hmm, I guess this actually bans the usage of any function named `require` ever, but I'm not sure there's any better way to do it.

File: src/rules/noRequireImportsRule.ts [hunk 3]
```diff
@@ -26,27 +26,20 @@ export class Rule extends Lint.Rules.AbstractRule {
...
+        if (node.arguments != null && node.expression != null) {
+            const callExpressionText = node.expression.getText(this.getSourceFile());
+            if (callExpressionText === "require") {
+                this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.FAILURE_STRING));
+            }
         }
-        super.visitVariableStatement(node);
```
Comment (JKillian) [inline src/rules/noRequireImportsRule.ts]: We might still want to call super here, imagine something weird like:
```typescript
let foo;
let bar = someMethod(foo = require('foo'));
```

File: src/rules/noRequireImportsRule.ts (hunk 1 above)
Comment (adidahiya) [inline src/rules/noRequireImportsRule.ts]: no

File: src/rules/noRequireImportsRule.ts (hunk 2 above)
Comment (adidahiya) [inline src/rules/noRequireImportsRule.ts]: yeah, I would consider that bad form if you're not in a node environment. and it's simpler for TSLint not to ask about your project's environment (browser vs. node)

File: src/rules/noRequireImportsRule.ts (hunk 3 above)
Comment (adidahiya) [inline src/rules/noRequireImportsRule.ts]: ok good point

**Comment by adidahiya** [comment]
looks like this will not build until #834 goes in.

**Comment by adidahiya** [comment]
rebased & squashed.

**Pull request merged**